### PR TITLE
Add search field above campaigns list

### PR DESCRIPTION
**Pull request by KaliedaRik**
## What does this change?
Adds a search input field above the list of campaigns on the Campaign page.

This will help users more quickly find the campaign they are interested in

## Screenshots

**Before**
Screenshot 2022-11-28 at 14 17 50

**After**
Screenshot 2022-11-28 at 17 00 08
Screenshot 2022-11-28 at 17 00 24

### CODE REVIEW

File: public/src/components/channelManagement/campaigns/CampaignsList.tsx [hunk 1]
```diff
@@ -56,8 +58,16 @@ const CampaignsList = ({
     return 'outlined';
   };
 
-  // When we get creation timestamps, we are thinking of allowing users to sort the list either alphabetically or by most recently created. For now, only the alphabetically sorted version is possible
-  const sortedCampaigns = [...campaigns];
+  const sortedCampaigns = campaigns.filter(c => {
```
Comment (tomrf1) [inline public/src/components/channelManagement/campaigns/CampaignsList.tsx:61]: The `sortedCampaigns` name is quite confusing now that we're initialising it with a filter!
Perhaps the filtering could be moved to a function and then you can do something like:
`const sortedAndFilteredCampaigns = filterCampaigns().sort((a, b)...`

Comment (KaliedaRik) [reply public/src/components/channelManagement/campaigns/CampaignsList.tsx:61]: I'll give it a go

Comment (KaliedaRik) [reply public/src/components/channelManagement/campaigns/CampaignsList.tsx]: The Array `sort` function mutates the array so I've separated that out to its own function alongside the new filter function. Code now self-documents what is happening here through function/variable names

Comment (tomrf1) [reply public/src/components/channelManagement/campaigns/CampaignsList.tsx]: 👍

File: public/src/components/channelManagement/campaigns/CampaignsList.tsx [hunk 2]
```diff
@@ -56,21 +58,36 @@ const CampaignsList = ({
...
-  });
+      if (A < B) {
+        return -1;
+      }
+      if (B < A) {
+        return 1;
+      }
+      return 0;
+    });
+    return [...campaignArray];
```
[the remark this answers is not in the document]
Comment (KaliedaRik) [reply public/src/components/channelManagement/campaigns/CampaignsList.tsx]: Fixed

File: public/src/components/channelManagement/campaigns/CampaignsList.tsx [hunk 3]
```diff
@@ -56,21 +58,36 @@ const CampaignsList = ({
     return 'outlined';
   };
 
-  // When we get creation timestamps, we are thinking of allowing users to sort the list either alphabetically or by most recently created. For now, only the alphabetically sorted version is possible
-  const sortedCampaigns = [...campaigns];
+  const filterCampaigns = (campaignArray: Campaigns) => {
+    return campaignArray.filter(c => {
+      if (!campaignSearch) {
```
Comment (tomrf1) [inline public/src/components/channelManagement/campaigns/CampaignsList.tsx:63]: not terribly important but if you move this `if (!campaignSearch)` check outside of the filter then the browser will not have to do pointless work every time `campaignSearch` is empty

Comment (KaliedaRik) [reply public/src/components/channelManagement/campaigns/CampaignsList.tsx:63]: It's a good idea!

**Pull request merged**